src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java

```diff
@@ -584,6 +584,34 @@ private void buildAcknowledgementOfServiceReceived(EventHistory.EventHistoryBuil
             responseIntentionType = caseData.getRespondent2ClaimResponseIntentionType().getLabel();
         }
 
+        //finding acknowledged date for the correct respondent in a 1v2 different solicitor scenario
+        MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
```

Comment:
Could this determination of acknowledgement date move to a private method or a static method in a separate class. It would make testing the conditional branches easy to test. Currently line 604-606 are not covered by tests.